test/util.go

```diff
@@ -4,20 +4,51 @@ import (
 	"context"
 	"fmt"
 	"strings"
+	"time"
 
 	testcluster "github.com/mesosphere/ksphere-testing-framework/pkg/cluster"
 	networkutils "github.com/mesosphere/ksphere-testing-framework/pkg/utils/networking"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+const (
+	timeTimeout = 300 * time.Second
+)
+
 func portForwardPodWithPrefix(cluster testcluster.Cluster, ns, prefix, port string) (int, chan struct{}, error) {
 	pod, err := findPodWithPrefix(cluster, ns, prefix)
 	if err != nil {
 		return 0, nil, fmt.Errorf("could not find pod with prefix %s: %s", prefix, err)
 	}
-	if pod.Status.Phase != corev1.PodRunning {
-		return 0, nil, fmt.Errorf("pod %s is not running, it's in phase %s", pod.Name, pod.Status.Phase)
+
+	// Wait for Pod running and its containers to be ready for some time or timeout!
+	timeout := time.After(timeTimeout)
+	tick := time.Tick(500 * time.Millisecond)
+	var containersReady bool
+	for {
+		select {
+		case <-timeout:
+			return 0, nil, fmt.Errorf("Timeout after %.0f minutes - Pod %s in phase %s, containers of it are overall not ready", timeTimeout.Minutes(), pod.Name, pod.Status.Phase)
+		case <-tick:
+			// Check first if Pod is NOT in Running phase
+			if pod.Status.Phase != corev1.PodRunning {
+				continue
+			}
+			// Check second if containers are NOT in READY phase
+			for _, c := range pod.Status.ContainerStatuses {
+				if !c.Ready && containersReady {
+					containersReady = false
+				} else {
+					containersReady = true
+				}
+			}
+			if !containersReady {
+				continue
+			}
+			break
+		}
+		break
```

Comment:
Waiting for the pod phase to be running like this seems like a generally good improvement :+1: